common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -517,12 +518,15 @@ private void useSmartcardCertForAuth(@NonNull final X509Certificate cert,
             final X509Certificate[] chain = new X509Certificate[]{cert};
             //Clear current dialog.
             mDialogHolder.dismissDialog();
+            //Set mIsSmartcardCertBasedAuthProceeding to true so a result will be telemeterized.
+            mIsSmartcardCertBasedAuthProceeding = true;
             //Call proceed on ClientCertRequest with PivPrivateKey and cert chain.
             request.proceed(pivPrivateKey, chain);
         } catch (final UnrecoverableKeyException | CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException e) {
             Logger.error(methodTag, e.getMessage(), e);
             //Show general error dialog.
             mDialogHolder.showErrorDialog(R.string.smartcard_general_error_dialog_title, R.string.smartcard_general_error_dialog_message);
+            //TODO: Telemet why it was cancelled.
```

Comment:
note: this needs to be a separate event names from the one emitted in telemetCertBasedAuthResults, otherwise the code will emit an error twice (here, and when the webview requests actually fail), which could be confusing / skew the data.

Reply:
have to be honest, I forgot that I needed to emit telemetry for when the flow is cancelled (not proceeded yet). 
I'm considering creating a separate event name for cancelling.
Should I also be emitting telemetry for when the cancellation is not an error (ex. user hits 'cancel' button, has no certs loaded on key, is locked out from key)?